language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/DefinedDataType.java

```diff
@@ -28,7 +28,11 @@ public interface DefinedDataType<T> {
   /** Produce the encoded form. */
   Value toValue();
 
-  JsonLfEncoder jsonEncoder();
+  // To avoid breaking code generated by old code-generators that link against updated java-bindings
+  // we provide a default implementation. Calling toJson will fail but old code-gen doesn't have it.
+  default JsonLfEncoder jsonEncoder() {
+    return null;
```

Comment:
Nit: I'd throw `UnsupportedOperationException` instead of returning `null`. This also allows to put the explanation for failing into the exception `message`. If you return `null` here then it will fail down the road somewhere with a NPE which may be difficult to understand what the problem actually is.

Reply:
Done